debug/org.eclipse.debug.ui/ui/org/eclipse/debug/internal/ui/actions/AbstractEnableAllActionDelegate.java

```diff
@@ -0,0 +1,86 @@
+package org.eclipse.debug.internal.ui.actions;
+
+import org.eclipse.jface.action.IAction;
+import org.eclipse.jface.viewers.ISelection;
+import org.eclipse.swt.widgets.Event;
+import org.eclipse.ui.IActionDelegate2;
+import org.eclipse.ui.IViewActionDelegate;
+import org.eclipse.ui.IViewPart;
+import org.eclipse.ui.IWorkbenchWindow;
+import org.eclipse.ui.IWorkbenchWindowActionDelegate;
+
+public abstract class AbstractEnableAllActionDelegate
```

Comment:
since there is a only a single implementation "EnableAllBreakpointsAction" it seems boiler plate code to have an AbstractEnableAllActionDelegate()

Reply:
Had the same thought too, but Disable All follows the same approach, so only tried the same.

Reply:
Had the same thought too, but Disable All & Remove all  follows the same approach, so only tried the same for Enable all. Can be used for Enable all actions for future components.